### DISCUSSION TITLE
Audio Lag

### DISCUSSION THREAD

**Discussion by chrischris1home** [mozilla/hubs#2856] [General Questions & Discussion] [answered]
Hi, reposting from Discord  here ..[https://discord.com/channels/498741086295031808/535638724785078292/743489890766946407].

I've been playing with hubs and love it! I see lots of applications at work. The one challenge I'm having is audio lag. I'm running two sessions, one on an ipad one on a windows 10 laptop, both based in the UK, and the audio lag is well over a second. Is there anything that can be optimised for this? Deploying into a UK DC using hubs cloud might, but the use cases I'm looking at are global so more interested in ways to optimise audio globally rather than for UK. 

Thanks for any help anyone can give!

**Comment by InfiniteLee** [comment, marked as answer]
https://hubs.mozilla.com's servers are all hosted in AWS' Northern California data center. As a result, much of the lag you're experiencing is latency with those servers- all data sent from a user has to be sent to our servers, then out to the other users. I think roughly that equates to ~250ms latency (just from transit) if going from the UK to California, back to the UK. That alone wouldn't warrant over a second of lag, but local network conditions may further exacerbate it.

Ideally you should run a Hubs Cloud server in a location that is central to as much of your users as you can predict, but I would start by running one in the UK and seeing how much latency you're getting there, just to rule out any local network conditions as a factor.
